File: stack/739.Daily Temperatures/739.go
func dailyTemperatures(temperatures []int) []int {
    n := len(temperatures)
    result := make([]int, n)  
    stack := []int{}          
    
    for i := 0; i < n; i++ {
        // Mevcut sıcaklık, stack'teki sıcaklıklardan büyükse
        for len(stack) > 0 && temperatures[stack[len(stack)-1]] < temperatures[i] {
            prevIndex := stack[len(stack)-1]
            stack = stack[:len(stack)-1]  // pop
            result[prevIndex] = i - prevIndex
        }
        
        stack = append(stack, i)
    }
    
    return result
}

File: stack/739.Daily Temperatures/739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures)
    {
        int n = temperatures.size();
        vector<int> result(n, 0); //Başlangıçta tüm elemanları 0 ile başlattık.
        stack<int> st; // Indexleri tutacak olan stack.

        for (int i=0; i<n; i++)
        {
            while (!st.empty() && temperatures[st.top()] < temperatures[i])
            {
                int prevIndex = st.top();
                st.pop();
                result[prevIndex] = i - prevIndex;
            }
            st.push(i);
        }
        return result;
    }
};
